config/config.go

```diff
@@ -12,594 +12,91 @@
-
-	prodEndpoint = "{zone}-osconfig.googleapis.com:443"
-
-	osInventoryEnabledDefault      = false
-	guestPoliciesEnabledDefault    = false
-	taskNotificationEnabledDefault = false
-	debugEnabledDefault            = false
-
-	configDirWindows     = `C:\Program Files\Google\OSConfig`
-	configDirLinux       = "/etc/osconfig"
-	taskStateFileWindows = configDirWindows + `\osconfig_task.state`
-	taskStateFileLinux   = configDirLinux + "/osconfig_task.state"
-	restartFileWindows   = configDirWindows + `\osconfig_agent_restart_required`
-	restartFileLinux     = configDirLinux + "/osconfig_agent_restart_required"
-
-	osConfigPollIntervalDefault = 10
-	osConfigMetadataPollTimeout = 60
-	osConfigWatchConfigTimeout  = 10 * time.Minute
-)
-
-var (
-	endpoint = flag.String("endpoint", prodEndpoint, "osconfig endpoint override")
-	debug    = flag.Bool("debug", false, "set debug log verbosity")
-	stdout   = flag.Bool("stdout", false, "log to stdout")
-
-	agentConfig   = &config{}
-	agentConfigMx sync.RWMutex
-	version       string
-	lEtag         = &lastEtag{Etag: "0"}
-
-	// Current supported capabilites for this agent.
-	// These are matched server side to what tasks this agent can
-	// perform.
-	capabilities = []string{"PATCH_GA", "GUEST_POLICY_BETA"}
-)
-
-type config struct {
-	osInventoryEnabled, guestPoliciesEnabled, taskNotificationEnabled, debugEnabled       bool
-	svcEndpoint, googetRepoFilePath, zypperRepoFilePath, yumRepoFilePath, aptRepoFilePath string
-	numericProjectID, osConfigPollInterval                                                int
-	projectID, instanceZone, instanceName, instanceID                                     string
-}
-
-func (c *config) parseFeatures(features string, enabled bool) {
-	for _, f := range strings.Split(features, ",") {
-		f = strings.ToLower(strings.TrimSpace(f))
-		switch f {
-		case "tasks", "ospatch": // ospatch is the legacy flag
-			c.taskNotificationEnabled = enabled
-		case "guestpolicies", "ospackage": // ospackage is the legacy flag
-			c.guestPoliciesEnabled = enabled
-		case "osinventory":
-			c.osInventoryEnabled = enabled
-		}
-	}
-}
-
-func (c *config) asSha256() string {
-	h := sha256.New()
-	h.Write([]byte(fmt.Sprintf("%v", c)))
-
-	return fmt.Sprintf("%x", h.Sum(nil))
-}
-
-func getAgentConfig() config {
-	agentConfigMx.RLock()
-	defer agentConfigMx.RUnlock()
-	return *agentConfig
+	managedResources *ManagedResources
+	inDS             bool
 }
 
-type lastEtag struct {
-	mu   sync.RWMutex
-	Etag string
+// InDesiredState reports whether this resource is in the desired state.
+// CheckState or EnforceState should be run prior to calling InDesiredState.
+func (r *Resource) InDesiredState() bool {
+	return r.inDS
 }
 
-func (e *lastEtag) set(etag string) {
-	e.mu.Lock()
-	defer e.mu.Unlock()
-	e.Etag = etag
+// ManagedResources returns the resources that this resources manages.
+func (r *Resource) ManagedResources() *ManagedResources {
```

Comment:
Hmm, what does this mean? Do resources manage other resources?

Reply:
Wasn't sure of a better name here, ManagedItems maybe? this is the actual objects this resource manages like packages, files or other item.

Now that I renamed Resource to OSPolicyResource this may not be as confusing.